src/native/s3_client.c

```diff
@@ -404,6 +406,14 @@ JNIEXPORT jlong JNICALL Java_software_amazon_awssdk_crt_s3_S3Client_s3ClientMake
         aws_s3_init_default_signing_config(signing_config, region, credentials_provider);
     }
 
+    struct aws_uri endpoint;
+    AWS_ZERO_STRUCT(endpoint);
+    if (jni_endpoint != NULL) {
+        struct aws_byte_cursor endpoint_str = aws_jni_byte_cursor_from_jbyteArray_acquire(env, jni_endpoint);
+        AWS_FATAL_ASSERT(aws_uri_init_parse(&endpoint, allocator, &endpoint_str) == AWS_OP_SUCCESS);
```

Comment:
Well, I think we can handle this error. That should not crash the program. 
```suggestion
        if (aws_uri_init_parse(&endpoint, allocator, &endpoint_str)) {
            aws_jni_throw_runtime_exception(env, "S3Client.aws_s3_client_make_meta_request: failed to parse endpoint");
            aws_jni_byte_cursor_from_jbyteArray_release(env, jni_endpoint, endpoint_str);
            goto done;
        }
```

Also, for proper clean up, you need to move the endpoint part after `request_message` created, as we clean up the resources from `done` tag

Reply:
done - handled the error  very slightly differently to call `aws_jni_byte_cursor_from_jbyteArray_release` only once